src/gui/editlocallyjob.cpp

```diff
@@ -331,22 +331,23 @@ void EditLocallyJob::lockFile()
         return;
     }
 
-    _folderConnections.append(connect(_accountState->account().data(), &Account::lockFileSuccess,
-                                      this, [this] {
-        _folderForFile->journalDb()->schedulePathForRemoteDiscovery(_relPath);
-        _folderForFile->scheduleThisFolderSoon();
-    }));
-    _folderConnections.append(connect(_folderForFile, &Folder::syncFinished,
-                                      this, [this](const OCC::SyncResult &result) {
-        Q_UNUSED(result)
-        fileLockSuccess();
+    _folderForFile->setSyncPaused(true);
+
+    _folderConnections.append(connect(&_folderForFile->syncEngine(), &SyncEngine::itemCompleted,
+                                      this, [this](const SyncFileItemPtr &item) {
+        if (item->_file == _relPath && item->_locked == SyncFileItem::LockStatus::LockedItem) {
```

Comment:
@claucambra Have you tested this with the scenario of setting a folder connection to a subfolder on a remote server? E.G., a server has folders `A, B, C`, and you add a folder connection pointing to a remote path `/B`. Will this then work? The `const SyncFileItemPtr &item` will likely have its `_file` member set to just `fileName.extension`, without the `/B` part. While the `_relPath` will be something like `/B/fileName.extension` as per the remote path. The local database does not account for the local folder being connected to a subfolder of a remote root and always considers top-level items as being in the root folder. In that case, this comparison will not work for such folders.